sklearn/svm/tests/test_svm.py

```diff
@@ -125,7 +126,7 @@ def test_precomputed():
 
     kfunc = lambda x, y: np.dot(x, y.T)
     clf = svm.SVC(kernel=kfunc)
-    clf.fit(X, Y)
+    clf.fit(np.array(X), Y)
```

Comment:
basically here it's an API change. Before for a custom kernel we would convert X to array. I propose to delegate the input checking to kernel function.